processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -1125,6 +1162,19 @@ public ColumnCapabilitiesImpl getCapabilities()
       this.rowIndex = rowIndex;
     }
 
+    TimeAndDims(
```

Comment:
This constructor is dangerously close in signature to the other 4-arg constructor. The only difference is an `int` vs a `long`, and those can get implicitly cast to each other, making calling code harder to read and harder to write correctly. It'd be better to either rework the constructors to be less conflicty (perhaps add `dimsKeySize` to all of them) or to instead go with the private-constructor-plus-static-creator-methods pattern. The static creator methods can have descriptive names that clarify them.

Reply:
Good catch and nice tip to use private-constructor-plus-static-creator-methods pattern.